astropy/table/serialize.py

```diff
@@ -75,13 +75,13 @@ def _represent_mixin_as_column(col, name, new_cols, mixin_cols,
     # - description: DO store
     # - meta: DO store
     info = {}
-    for attr, nontrivial, xform in (('unit', lambda x: x is not None and x != '', str),
-                                    ('format', lambda x: x is not None, None),
-                                    ('description', lambda x: x is not None, None),
-                                    ('meta', lambda x: x, None)):
+    for attr, nontrivial in (('unit', lambda x: x is not None and x != ''),
+                             ('format', lambda x: x is not None),
+                             ('description', lambda x: x is not None),
+                             ('meta', lambda x: x)):
         col_attr = getattr(col.info, attr)
         if nontrivial(col_attr):
-            info[attr] = xform(col_attr) if xform else col_attr
+            info[attr] = col_attr
```

Comment:
In the end, the main thing was not to convert the unit to a string prematurely - this then later changed a user-defined unit into `UnrecognizedUnit` (in the creation of a `Column(..., **info)`, which was undesired. And it is not needed for the `yaml` storing.